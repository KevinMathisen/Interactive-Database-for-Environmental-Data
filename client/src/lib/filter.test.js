import { render, fireEvent } from '@testing-library/svelte';
import { describe, expect, it } from 'vitest';
import Filter from './Filter.svelte';

/**
 * @vitest-environment jsdom
 */

describe('Filter', () => {
  it('species loads correctly', async () => {
    const selectableSpecies = ['species1', 'species2', 'species3'];

    const { component, getByRole } = render(Filter, { props: {selectableSpecies} });

    const list = getByRole('list');

    expect(list.childElementCount).toEqual(3);

  });
});