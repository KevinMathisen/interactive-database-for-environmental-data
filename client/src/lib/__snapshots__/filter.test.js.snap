// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Filter > renders correctly 1`] = `
<body>
  <div>
    <div
      class="main s-_LI_r-dATrBW"
    >
      <form
        class="s-_LI_r-dATrBW"
      >
        <div
          class="collapsibleSection s-qco_XiUj5iHx"
        >
          <div
            class="collapsibleSection-header s-qco_XiUj5iHx"
            role="button"
            tabindex="0"
          >
            <i
              class="arrow s-qco_XiUj5iHx not-collapsed"
            />
             
            <h3
              class="s-qco_XiUj5iHx"
            >
              Type data
            </h3>
          </div>
           
          <div
            class="content s-qco_XiUj5iHx"
          >
            <label
              class="s-I5YWuR1l8gCU"
              for="river"
            >
              <input
                class="s-I5YWuR1l8gCU"
                id="river"
                name="dataType"
                type="radio"
                value="river"
              />
               
              Elvedata
               
            </label>
            <label
              class="s-I5YWuR1l8gCU"
              for="station"
            >
              <input
                class="s-I5YWuR1l8gCU"
                id="station"
                name="dataType"
                type="radio"
                value="station"
              />
               
              Stasjonsdata
               
            </label>
            
            <!--&lt;RadioInput&gt;-->
          </div>
        </div>
        <!--&lt;CollapsibleSection&gt;-->
         
        <div
          class="collapsibleSection s-qco_XiUj5iHx"
        >
          <div
            class="collapsibleSection-header s-qco_XiUj5iHx"
            role="button"
            tabindex="0"
          >
            <i
              class="arrow s-qco_XiUj5iHx not-collapsed"
            />
             
            <h3
              class="s-qco_XiUj5iHx"
            >
              Dato
            </h3>
          </div>
           
          <div
            class="content s-qco_XiUj5iHx"
          >
            <label
              class="s-JXk2os4N_waP"
              for="startDate"
            >
              Fra
  
              <input
                class="s-JXk2os4N_waP"
                id="startDate"
                name="startDate"
                type="date"
              />
            </label>
             
            <label
              class="s-JXk2os4N_waP"
              for="endDate"
            >
              Til
  
              <input
                class="s-JXk2os4N_waP"
                id="endDate"
                name="endDate"
                type="date"
              />
            </label>
            <!--&lt;DateInput&gt;-->
          </div>
        </div>
        <!--&lt;CollapsibleSection&gt;-->
         
        <div
          class="collapsibleSection s-qco_XiUj5iHx"
        >
          <div
            class="collapsibleSection-header s-qco_XiUj5iHx"
            role="button"
            tabindex="0"
          >
            <i
              class="arrow s-qco_XiUj5iHx not-collapsed"
            />
             
            <h3
              class="s-qco_XiUj5iHx"
            >
              Art
            </h3>
          </div>
           
          <div
            class="content s-qco_XiUj5iHx"
          >
            <label
              class="s-a2sscjfPS7lx"
              for="all"
            >
              <input
                class="s-a2sscjfPS7lx"
                id="all"
                name="all"
                type="radio"
                value="true"
              />
               Velg alle
            </label>
             
            <label
              class="s-a2sscjfPS7lx"
              for="custom"
            >
              <input
                class="s-a2sscjfPS7lx"
                id="custom"
                name="custom"
                type="radio"
                value="false"
              />
               Egendefinert
            </label>
             
            
            <!--&lt;SpeciesInput&gt;-->
          </div>
        </div>
        <!--&lt;CollapsibleSection&gt;-->
      </form>
    </div>
    <!--&lt;Filter&gt;-->
  </div>
</body>
`;
